### PR TITLE
DPAD-528: Bugs in new Fandom Video Player

### PR DESCRIPTION
**Pull request by gundams-and-dragons**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@fandom/jwplayer-fandom",
-	"version": "2.0.0-beta-6",
+	"version": "2.0.0-dpad-528-13",
```
Comment (gundams-and-dragons) [inline package.json:3]: I'll change this version to the next before merge. Leaving as is until then as that's likely to change.

File: src/players/shared/JwPlayerWrapper.tsx [hunk 2]
```diff
@@ -61,6 +65,10 @@ const JwPlayerWrapper: React.FC<JwPlayerWrapperProps> = ({ playlist, playerUrl,
 					player: window.jwplayer(elementId),
 					ready: event,
 				});
+
+				if (onReady) {
```
Comment (p-forfa) [inline src/players/shared/JwPlayerWrapper.tsx:69]: Does this delay the loading of the video player?

File: src/players/shared/JwPlayerWrapper.tsx [hunk 3]
```diff
@@ -90,4 +98,4 @@ const JwPlayerWrapper: React.FC<JwPlayerWrapperProps> = ({ playlist, playerUrl,
 	);
 };
 
-export default JwPlayerWrapper;
+export default React.memo(JwPlayerWrapper);
```
Comment (p-forfa) [inline src/players/shared/JwPlayerWrapper.tsx:101]: 👍

File: src/utils/articleVideo/articlePlayerOnReady.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { communicationService } from 'utils/communication';
+import { setVideoSeenInSession } from 'utils/articleVideo/articleVideoSession';
+import { willAutoplay, willMute } from 'utils/articleVideo/articleVideoConfig';
+
+export default function useOnArticlePlayerReady(videoDetails, playerInstance): void {
+	const playerKey = 'aeJWPlayerKey';
```
Comment (p-forfa) [inline src/utils/articleVideo/articlePlayerOnReady.ts:6]: Is there any significance to the 'ae' prefix?

File: src/utils/articleVideo/articlePlayerOnReady.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		},
+		options: {
+			audio: !willMute(),
+			ctp: !willAutoplay(),
+			slotName: 'featured',
+			videoId: videoDetails.playlist[0].mediaid,
+		},
+	});
+
+	// player.on('autoplayToggle', function (data) {
```
Comment (p-forfa) [inline src/utils/articleVideo/articlePlayerOnReady.ts:31]: Is this meant to be commented out?

**Review by p-forfa — APPROVED** [review]
LGTM. Left a few comments around commented out code. Also, as mentioned the version in the package.json should be changed as well.

File: src/players/shared/JwPlayerWrapper.tsx (hunk 2 above)
Comment (gundams-and-dragons) [reply src/players/shared/JwPlayerWrapper.tsx:69]: No, at this point the player is "ready" and should be loaded. This is very similar to the current player's implementation.

File: src/utils/articleVideo/articlePlayerOnReady.ts (hunk 4 above)
Comment (gundams-and-dragons) [reply src/utils/articleVideo/articlePlayerOnReady.ts:6]: I think it stands for ad engine. The ad engine code checks for that one specifically .

File: src/utils/articleVideo/articleVideoConfig.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import * as articleVideoCookieService from 'utils/articleVideo/articleVideoCookies';
+import { wikiaJWPlayeri18n } from 'i18n';
+
+const isFromRecirculation = () => {
+	return window.location.search.indexOf('wikia-footer-wiki-rec') > -1;
```
[the remark this answers is not in the document]
Comment (gundams-and-dragons) [reply src/utils/articleVideo/articleVideoConfig.ts:5]: It is set externally. After release we should get together with the adengine team to lock down the specifics. For now, getting it to work as it currently does is my highest priority .

**Pull request merged**